lib/backup/notifier/command.rb

```diff
@@ -0,0 +1,99 @@
+# encoding: utf-8
+
+module Backup
+  module Notifier
+    class Command < Base
+
+      ##
+      # Command to execute.
+      #
+      # Make sure it is accessible from your $PATH, or provide
+      # the absolute path to the command.
+      attr_accessor :command
+
+      ##
+      # Arguments to pass to the command.
+      #
+      # Must be an array of strings or callable objects.
+      #
+      # Callables will be invoked with #call(model, status),
+      # and the return value used as the argument.
+      #
+      # In strings you can use the following placeholders:
+      #
+      # %l - Model label
+      # %t - Model trigger
+      # %s - Status (success/failure/warning)
+      # %v - Status verb (succeeded/failed/succeeded with warnings)
+      #
+      # All placeholders can be used with uppercase letters to capitalize
+      # the value.
+      #
+      # Defaults to ["%L %v"]
+      attr_accessor :args
+
+      def initialize(model, &block)
+        super
+        instance_eval(&block) if block_given?
+
+        @args ||= ["%L %v"]
+      end
+
+      private
+
+      ##
+      # Notify the user of the backup operation results.
+      #
+      # `status` indicates one of the following:
+      #
+      # `:success`
+      # : The backup completed successfully.
+      # : Notification will be sent if `on_success` is `true`.
+      #
+      # `:warning`
+      # : The backup completed successfully, but warnings were logged.
+      # : Notification will be sent if `on_warning` or `on_success` is `true`.
+      #
+      # `:failure`
+      # : The backup operation failed.
+      # : Notification will be sent if `on_warning` or `on_success` is `true`.
+      #
+      def notify!(status)
+        IO.popen([@command] + args.map { |arg| format_arg(arg, status) })
```

Comment:
I thought it would be something like that. It's not a custom notifier, but a built in one where you can't control the arguments. That's unfortunate and yes you would have to resort to the string replacement way of adding your arguments. I'll have a look this weekend and see how that'll fit in the way I implemented this in v5.

This PR looks fine otherwise and I'll look into merging it this week. If I find anything I'll let you know.